### DISCUSSION TITLE
Adding Firmware information extra sensors to the integration

### DISCUSSION THREAD

**Discussion by Rill189** [nathanmarlor/foxess_modbus#619] [General]
Love the integration,
There is quite a bit more information available through the RS485 such as firmware versions for the batteries, invertors etc. Could that be added?

Thanks

**Comment by canton7** [comment]
It could, but is it worth going to the effort?

**Comment by Rill189** [comment]
For trouble shooting it is worth noting the firmware versions to ensure they are the latest or at least the same.

Thanks great work :)

**Reply by canton7** [reply]
We do expose the firmware versions for the manager, master, and slave, for what it's worth. Those are the versions which the user normally cares about upgrading.

**Comment by Rill189** [comment]
I have specific issues with capacity which would make it really useful.

Thanks

**Reply by canton7** [reply]
Please reply using the github web interface -- the email reply functionality is pretty broken (as you'll see if you look at the web interface!).

Which versions specifically do you need to look at?

**Reply by Rill189** [reply]
Apologies,
Not sure what is available. But at least the battery modules individually and BMS versions.

Thanks
